### PR TITLE
gap: adapt get_global to use official libgap API

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

**Comment by fchapoton** [comment]
well, does not compile

**Comment by fchapoton** [comment]
I guess `GAP_ValueGlobalVariable` should be added to the appropriate `.pxd` file ?

**Comment by fingolfin** [comment]
Huh, `GAP_ValueGlobalVariable` is already in `gap_includes.pxd`, and in fact is already being used in `element.pyx` and `util.pyx`. Both of these, as well as `libgap.pyx` (which this PR modifies) also do this:
```python
from .gap_includes cimport *
```
So I have no idea what the problem is, I am afraid -- my Python / Cython knowledge is just too limited :-(

File: src/sage/libs/gap/libgap.pyx [hunk 1]
```diff
@@ -539,8 +539,7 @@ class Gap(Parent):
             ...
             GAPError: Error, VAL_GVAR: No value bound to FooBar
         """
-        value_global = self.function_factory('ValueGlobal')
-        return value_global(variable)
+        return make_any_gap_element(GAP_ValueGlobalVariable(variable.value))
```
Comment (fchapoton) [inline src/sage/libs/gap/libgap.pyx]: maybe using `return make_any_gap_element(self, GAP_ValueGlobalVariable(variable.value))`

**Comment by fchapoton** [comment]
now complains about
```
        return make_any_gap_element(self, GAP_ValueGlobalVariable(variable.value))
    AttributeError: 'str' object has no attribute 'value'
```

**Comment by fchapoton** [comment]
Now

      File "sage/libs/gap/libgap.pyx", line 542, in sage.libs.gap.libgap.Gap.get_global
        return make_any_gap_element(self, GAP_ValueGlobalVariable(variable))
    TypeError: expected bytes, str found

so we need to change the string to bytes. Using `str_to_bytes` ?

`from sage.cpython.string cimport str_to_bytes`

**Comment by fingolfin** [comment]
Using `str_to_bytes` sounds sensible. Thank you for your help.

**Comment by fchapoton** [comment]
I have tested locally, and it seems to work.

**Comment by fingolfin** [comment]
Fine by me

**Comment by fchapoton** [comment]
checks are ok. Shall I approve and set to positive ?

**Comment by fingolfin** [comment]
Fine by me

**Review by fchapoton — APPROVED** [review]
good enough, hopefully

**Pull request merged**